packages/task/src/browser/quick-open-task.ts

```diff
@@ -721,3 +721,96 @@ export class TaskRunningQuickOpen implements QuickOpenModel {
         });
     }
 }
+
+export class TaskRestartRunningQuickOpenItem extends QuickOpenItem {
+
+    constructor(
+        protected readonly task: TaskConfiguration,
+        protected readonly taskInfo: TaskInfo,
+        protected readonly taskService: TaskService,
+        protected readonly taskDefinitionRegistry: TaskDefinitionRegistry,
+        protected readonly labelProvider: LabelProvider,
+        protected readonly isMulti: boolean,
+        public readonly options: QuickOpenGroupItemOptions,
+    ) {
+        super(options);
+    }
+
+    getLabel(): string {
+        return this.taskInfo.config.label;
+    }
+
+    getDescription(): string {
+        if (!this.isMulti) {
+            return '';
+        }
+        if (this.taskDefinitionRegistry && !!this.taskDefinitionRegistry.getDefinition(this.task)) {
+            if (this.task._scope) {
+                return this.labelProvider.getName(new URI(this.task._scope));
+            }
+            return this.task._source;
+        } else {
+            return this.labelProvider.getName(new URI(this.task._source));
+        }
```

Comment:
I am not sure if we could call `TaskSourceResolver.resolve()` to make this function smaller. Could you please check ? Thank you !

Reply:
I updated the code to your suggestion, please let me know if it is fine :)

Reply:
yes it looks good :)